### PR TITLE
[lld] Sort data chunks before code chunks on ARM64EC.

### PR DESCRIPTION
**Pull request by cjacek**
Depends on !70721.

I found this MSVC behavior while investigating !69100. This will allow simplifying that PR.

### CODE REVIEW

File: lld/test/COFF/arm64ec-codemap.test [hunk 1]
```diff
@@ -110,6 +110,40 @@ DISASMM-NEXT:                 ...
...
+
+RUN: llvm-readobj --coff-load-config testdm.dll | FileCheck -check-prefix=CODEMAPDM %s
+CODEMAPDM:      CodeMap [
+CODEMAPDM-NEXT:     0x2000 - 0x2008  ARM64EC
+CODEMAPDM-NEXT:     0x3000 - 0x3006  X64
+CODEMAPDM-NEXT:     0x5200 - 0x5208  ARM64EC
+CODEMAPDM-NEXT:     0x6000 - 0x6006  X64
+CODEMAPDM-NEXT: ]
+
+Merging code data into code sections causes data to be separated from the code when sorting chunks.
```
Comment (mstorsjo) [inline lld/test/COFF/arm64ec-codemap.test:124]: What's "code data" here, is there a typo in this sentence?

**Review by mstorsjo — APPROVED** [review]
LGTM overall, the change seems reasonable. One comment in the testcase had me scratching my head though.

File: lld/test/COFF/arm64ec-codemap.test (hunk 1 above)
Comment (cjacek) [reply lld/test/COFF/arm64ec-codemap.test]: It's indeed a typo, I meant "data section". I pushed a fixup, thanks for reviews!

**Comment by nico** [comment]
Looks like this broke check-lld: http://45.33.8.238/linux/122206/step_11.txt

Please take a look and revert for now if it takes a while to fix.

**Comment by cjacek** [comment]
> Looks like this broke check-lld: http://45.33.8.238/linux/122206/step_11.txt
> 
> Please take a look and revert for now if it takes a while to fix.

Sorry about that. I pushed a fixup as 12e39a322fe7757a3794e7425ee265f219d6dd2e.

I missed llvm-mc invocation when I was moving the test between branches and I didn't catch it because I had a stray object file in the tree.

**Pull request merged**